docs/docs/developers/tutorials/crowdfunding/donations.md

```diff
@@ -0,0 +1,202 @@
+
+### 2. Taking private donations
+
+#### Checking campaign duration against the timestamp
+
+To check that the donation occurs before the campaign deadline, we must access the public `timestamp`. It is one of several [Public Global Variables](https://docs.aztec.network/developers/contracts/references/globals#public-global-variables).
+
+Declare an Aztec function that is public and internal
+
+```rust
+#include_code deadline-header /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr raw
+    //...
+}
+```
+
+Read the deadline from storage and assert that the `timestamp` from this context is before the deadline
+
+#include_code deadline /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+---
+
+Since donations are to be private, the donate function will have the user's private context which has these [Private Global Variables](https://docs.aztec.network/developers/contracts/references/globals#private-global-variables). So from the private context there is a little extra to call the (public internal) `_check_deadline` function.
+
+```rust
+#include_code call-check-deadline /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr raw
+  //...
+}
+```
+
+Namely calling `enqueue` and passing the (mutable) context.
+
+Now conclude adding all dependencies to the `Crowdfunding` contract:
+
+#include_code all-deps /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+Like before, you can find these and other `aztec::protocol_types` [here](https://github.com/AztecProtocol/aztec-packages/blob/#include_aztec_version/noir-projects/noir-protocol-circuits/crates/types/src).
+
+
+#### Interfacing with another contract
+
+The token being used for donations is stored simply as an `AztecAddress` (named `donation_token`). so to easily use it as a token, we let the compiler know that we want the address to have a Token interface. Here we will use a maintained example Token contract.
+
+Add this `Token` contract to Nargo.toml: `aztec = { git="https://github.com/AztecProtocol/aztec-packages/", tag="#include_aztec_version", directory="noir-projects/noir-contracts/contracts/token_contract" }`
+
+With the dependency already `use`d at the start of the contract, the token contract can be called to make the transfer from msg sender to this contract. Note: the user must have authorised this action (concept [here](../../../learn/concepts/accounts/main#authorizing-actions)), example use of `createAuthWit` in 'full donor flow' test [here](../../../../../yarn-project/end-to-end/src/e2e_crowdfunding_and_claim.test.ts).
+
+#### Creating and storing a private receipt note
+
+The last thing to do is create a new value note and add it to the `donation_receipts`. So the full donation function is now
+
+#include_code all-deps /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+### 3. Operator withdrawals
```

Comment:
I see the following error in the withdraw function

![image](https://github.com/AztecProtocol/aztec-packages/assets/18372439/50e5e36f-fb13-40de-9bea-b1a4e46e392b)